### PR TITLE
streamflow cartogram of July percentiles

### PR DESCRIPTION
**Pull request by collnell**
This builds off the data processing in https://github.com/USGS-VIZLAB/gage-conditions-gif using data generated for the last year (July 1st 2020-2021. Need to publicly serve up input data to create graphic.  

The pipeline produces this:
flow_cartogram-01

Which was refined +annotated outside of R for this final product:
flow_facet_july-01

### CODE REVIEW

File: flow_cartogram/src/prep_data.R [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+flow_by_day_by_state <- function(data_in, dv_site, sites_state) {
+  data_in  %>%
+    left_join(dv_site) %>% # adds state info for each gage
+    group_by(state_cd, date, cond) %>% # aggregate by state, day, flow condition
+    summarize(n_gage = length(unique(site_no)))  %>% 
+    left_join(sites_state) %>% # add total_gage
+    mutate(prop = n_gage/total_gage) %>% # proportion of gages
+    pivot_wider(id_cols = !n_gage, names_from = cond, values_from = prop, values_fill = 0) %>% # complete data for timepoints with 0 gages
```
Comment (lindsayplatt) [inline flow_cartogram/src/prep_data.R:48]: This is a neat lol' trick to fill those values :)

**Review by lindsayplatt — COMMENTED** [review]
Wow! This is awesome! Learning so much new fun stuff, like `patchwork` and `ggfx::with_shadow()`. 

When I run `tar_make()` (after getting the data in the folder), it errors out on the final step because the `out/` folder doesn't exist by default. I think you should fix this by either 

1. just having the image file be created top-level, or 
2. commit a ".placeholder" or ".empty" file (create a text file, put nothing in it, and save with those names but no ".txt" extension) in the `out/` folder so that when people use this repo, that folder already exists. There is an example of this [here](https://github.com/USGS-R/ds-pipelines-targets-2-template/tree/main/3_visualize/out).

If you don't do that, this is the error people will see:
```r
Error : cannot open stream out/flow_cartogram.svg
In addition: There were 14 warnings (use warnings() to see them)
Error: callr subprocess failed: cannot open stream out/flow_cartogram.svg
```

After I add the `out/` folder, I have success!

image

File: flow_cartogram/src/prep_data.R [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  data_in %>% 
+    mutate(date = as.Date(dateTime),
+           cond = 
+             case_when(
+               per < 0.05 ~ 'Driest',
+               per >= 0.05 & per < 0.1 ~ "Drier",
+               per >= 0.1 & per < 0.25 ~ "Dry", 
+               per >= 0.25 & per < 0.75 ~ "Normal",
+               per >= 0.75 & per < 0.9 ~ "Wet", 
+               per >= 0.9 ~ "Wettest"
```
[the remark this answers is not in the document]
Comment (collnell) [reply flow_cartogram/src/prep_data.R:11]: You did actually, I just pulled an earlier one when I was tinkering in the pipeline :)

File: flow_cartogram/src/plot_cartogram.R [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+combine_plots <- function(file_out, plot_left, plot_right, width, height, color_bknd){
+  
+  plot_blank <- ggplot()+
+    geom_blank()+ 
+    theme(panel.background = element_rect(fill = "blue",
+                                          color = color_bknd),
+          plot.background = element_rect(fill = color_bknd,
+                                         color = color_bknd))
+  
+  ((plot_blank / plot_left / plot_blank) | plot_right) +
```
[the remark this answers is not in the document]
Comment (lindsayplatt) [reply flow_cartogram/src/plot_cartogram.R:112]: oh can't wait!

**Comment by collnell** [comment]
I removed the `out/` folder.

This is my first venture with `ggfx`. It was also super easy!

**Review by lindsayplatt — APPROVED** [review]

**Pull request merged**